tools/apiview/parsers/js-api-parser/src/generate.ts

```diff
@@ -0,0 +1,254 @@
+import { buildToken, splitAndBuild } from "./jstokens";
+
+interface Metadata {
+  Name: string;
+  PackageName: string;
+  PackageVersion: string;
+  ParserVersion: string;
+  Language: "JavaScript";
+}
+
+// key: item's canonical reference, value: sub paths that include it
+const exported = new Map<string, Set<string>>();
+
+function emptyLine(relatedToLine?: string): ReviewLine {
+  return { RelatedToLine: relatedToLine, Tokens: [] };
+}
+
+function buildDependencies(reviewLines: ReviewLine[], dependencies: Record<string, string>) {
+  if (!dependencies) {
+    return;
+  }
+  const keys = Object.keys(dependencies);
+  if (keys.length === 0) {
+    return;
+  }
+
+  const header: ReviewLine = {
+    LineId: "Dependencies",
+    Tokens: [
+      buildToken({
+        Kind: TokenKind.StringLiteral,
+        Value: "Dependencies:",
+        NavigationDisplayName: "Dependencies",
+        RenderClasses: ["dependencies"],
+      }),
+    ],
+  };
+
+  const dependencyLines: ReviewLine[] = [];
+  for (const dependency of Object.keys(dependencies)) {
+    const nameToken: ReviewToken = buildToken({
+      Kind: TokenKind.StringLiteral,
+      Value: dependency,
+    });
+    const versionToken: ReviewToken = buildToken({
+      Kind: TokenKind.StringLiteral,
+      Value: ` ${dependencies[dependency]}`,
+      SkipDiff: true,
+    });
+    const dependencyLine: ReviewLine = {
+      Tokens: [nameToken, buildToken({ Kind: TokenKind.Punctuation, Value: ":" }), versionToken],
+    };
+    dependencyLines.push(dependencyLine);
+  }
+  header.Children = dependencyLines;
+  reviewLines.push(header);
+  reviewLines.push(emptyLine(header.LineId));
+}
+
+function buildSubpathExports(reviewLines: ReviewLine[], apiModel: ApiModel) {
+  for (const modelPackage of apiModel.packages) {
+    for (const entryPoint of modelPackage.entryPoints) {
+      const subpath = entryPoint.name.length > 0 ? entryPoint.name : "<default>";
+      const exportLine: ReviewLine = {
+        LineId: `Subpath-export-${subpath}`,
+        Tokens: [
+          buildToken({
+            Kind: TokenKind.StringLiteral,
+            Value: ` "${subpath}"`,
+            NavigationDisplayName: `"${subpath}" subpath export`,
+          }),
+        ],
+        Children: [],
+      };
+
+      for (const member of entryPoint.members) {
+        const canonicalRef = member.canonicalReference.toString();
+        const containingExport = exported.get(canonicalRef) ?? new Set<string>();
+        if (!containingExport.has(subpath)) {
+          containingExport.add(subpath);
+          exported.set(canonicalRef, containingExport);
+        }
+        buildMember(exportLine.Children!, member);
+      }
+      reviewLines.push(exportLine);
+    }
+  }
+}
+
+function buildDocumentation(reviewLines: ReviewLine[], member: ApiItem, relatedTo: string) {
+  if (member instanceof ApiDocumentedItem) {
+    const lines = member.tsdocComment?.emitAsTsdoc().split("\n");
```

Comment:
Does this add doc-comments to the APIView? 😍

Reply:
Yes based on my understanding.  It will add whatever ref docs included in the api.json. By default it is hidden.